appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/receiver/presenter/AccountInfoPresenter.java

```diff
@@ -0,0 +1,180 @@
+package org.kinecosystem.appsdiscovery.receiver.presenter;
+
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.support.annotation.IntDef;
+import android.support.annotation.NonNull;
+
+import org.jetbrains.annotations.NotNull;
+import org.kinecosystem.appsdiscovery.base.BasePresenter;
+import org.kinecosystem.appsdiscovery.receiver.manager.AccountInfoManager;
+import org.kinecosystem.appsdiscovery.receiver.manager.IAccountInfo;
+import org.kinecosystem.appsdiscovery.receiver.view.IAccountInfoView;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+public class AccountInfoPresenter extends BasePresenter<IAccountInfoView> implements IAccountInfoPresenter {
+    private static final String EXTRA_SOURCE_APP_NAME = "EXTRA_SOURCE_APP_NAME";
+    private static final int TASK_STATE_UNDEFINED = 0;
+    private static final int TASK_STATE_SUCCESS = 10;
+    private static final int TASK_STATE_FAILURE = 20;
+
+    @IntDef({TASK_STATE_UNDEFINED, TASK_STATE_SUCCESS, TASK_STATE_FAILURE})
+    @Retention(RetentionPolicy.SOURCE)
+    @interface TaskState {
+    }
+
+    private AccountInfoManager accountInfoManager;
+    private AccountInfoAsyncTask asyncTask;
+    private boolean isPaused = false;
+    @TaskState
+    private int taskState = TASK_STATE_UNDEFINED;
+    private Intent intent;
+
+    public AccountInfoPresenter(@NonNull AccountInfoManager accountInfoManager, @NonNull Intent intent) {
+        this.accountInfoManager = accountInfoManager;
+        this.intent = intent;
+    }
+
+    @Override
+    public void onAttach(@NotNull IAccountInfoView view) {
```

Comment:
accountInfoManager should be passed on the onAttach method - not at the constructor, because it is set to null at onDetach

Reply:
on second thought will add a separate start method ...